ownCloud/Migration/Migration.swift

```diff
@@ -0,0 +1,517 @@
+
+											// For the active account, migrate instant upload settings
+											if let activeAccount = rowDict["activeaccount"] as? Int, activeAccount == 1 {
+												self.migrateInstantUploadSettings(for: bookmark, userId: userId, accountDictionary: rowDict)
+											}
+
+											Log.debug(tagged: ["MIGRATION"], "Bookmark successfully added")
+
+											self.postAccountMigrationNotification(activity: bookmarkActivity, state: .finished, type: .account)
+
+										} else {
+											self.postAccountMigrationNotification(activity: bookmarkActivity, state: .failed, type: .account)
+										}
+									}
+								}
+
+							}, error: &error)
+						}
+					}
+
+					if let usersQuery = OCSQLiteQuery(selectingColumns: nil, fromTable: "users", where: nil, orderBy: nil, limit: nil, resultHandler: queryResultHandler) {
+						db.execute(usersQuery)
+					}
+
+					self.migrationQueue.async {
+						// Check if the passcode is set
+						let passcodeQuery = OCSQLiteQuery(selectingColumns: ["passcode"], fromTable: "passcode", where: nil, orderBy: "id DESC", limit: "1") { (_, _, _, resultSet) in
+							if let dict = try? resultSet?.nextRowDictionary(), let passcode = dict?["passcode"] as? String {
+
+								let activityName = "App Passcode".localized
+								self.postAccountMigrationNotification(activity: activityName, state: .initiated, type: .passcode)
+
+								Log.debug(tagged: ["MIGRATION"], "Migrating passcode lock")
+
+								if passcode.count == 4 && passcode.rangeOfCharacter(from: CharacterSet.decimalDigits.inverted) == nil {
+									AppLockManager.shared.passcode = passcode
+									AppLockManager.shared.lockEnabled = true
+									self.postAccountMigrationNotification(activity: activityName, state: .finished, type: .passcode)
+								} else {
+									self.postAccountMigrationNotification(activity: activityName, state: .failed, type: .passcode)
+									Log.error(tagged: ["MIGRATION"], "Passcode is invalid")
+								}
+							}
+						}
+						if let query = passcodeQuery {
+							db.execute(query)
+						}
+					}
+
+					self.migrationQueue.async {
+						DispatchQueue.main.async {
+							NotificationCenter.default.post(name: Migration.FinishedNotification, object: nil)
+						}
+					}
+
+				} else {
+					Log.error(tagged: ["MIGRATION"], "Couldn't open legacy database, error \(String(describing: err))")
+				}
+			}
+		}
+	}
+
+	// MARK: - Private helpers
+
+	private func setup(connection:OCConnection) -> [OCAuthenticationMethodIdentifier]? {
+		let connectGroup = DispatchGroup()
+		var supportedAuthMethods: [OCAuthenticationMethodIdentifier]?
+
+		func connectAndAuthorize() {
+			connectGroup.enter()
+			connection.prepareForSetup(options: nil) { (issue, _, supportedAuthenticationMethods, _) in
+				supportedAuthMethods = supportedAuthenticationMethods
+
+				if let issue = issue {
+					if issue.level != .error, supportedAuthMethods == nil {
+						issue.approve()
```

Comment:
@felix-schwarz yes, agree.. my assumption however was that if user was using the account in the legacy app, then It can be migrated AS IS. Also probably accepting a warning about using `http` is fine, but certificate acceptance is of course more sensitive topic, since we are not comparing here if it is same certificate which was accepted in the legacy app. 

So may be using similar mechanism as used during bookmark creation utilising `ConnectionIssueViewController` could be used and every warning should be re-accepted.

Reply:
@felix-schwarz I've implemented the handling I suggested.